crawl-ref/source/player.cc

```diff
@@ -4745,11 +4745,21 @@ void dec_channel_player(int delay)
 bool invis_allowed(bool quiet, string *fail_reason)
 {
     string msg;
+    string reason = "";
     bool success = true;
 
     if (you.haloed() && you.halo_radius() != -1)
     {
-        msg = "Your halo prevents invisibility.";
+        if (player_equip_unrand(UNRAND_EOS))
+            reason = "weapon";
```

Comment:
I would say "weapon's halo" or something like that, to explain why the weapon makes invisibility impossible. Maybe even change the "your ____ prevents invisibility" style to "Your weapon shines too brightly to become invisible".

Reply:
Great idea!